### PR TITLE
Track email form events

### PR DESCRIPTION
**Pull request by homer0**
## What does this PR do?

TL;DR: The email form now tracks whenever the user gets an error and when the address is successfully saved.

I first added a `utils` _"service"_ with two functions I needed in order to access the the global `ga` for the event tracking:

- `deferred`: To get a promise that can be resolved _externally_. It's basically a native port of [this](https://api.jquery.com/jquery.deferred/).
- `waitForGlobal`: Uses `setInterval` and a deferred promise to wait for a global variable. This is useful for 3rd parties libraries that are not installed or that can't be accessed through a module.

Then I added a `trackEvent` function that basically uses `waitForGlobal` to get the Analytics client (`ga`), with caching in the middle (of course), and uses it to send a custom custom event. This function only works on `process.env.NODE_ENV === 'production'` because that's the same condition the Gatsby plugin for GA has. On development, the function doesn't do anything.

Finally, I set the email form to track when there are validation/api errors and when the email is successfully saved.

PD: I added a tiny PR template, if anyone has an objection, I can remove it, no worries.

## How should it be tested manually?

First, let's test that this doesn't work on development:

```sh
yarn start
```

1. Go to `http://localhost:8000/`.
2. Open the networks tab.
2. Write an invalid email address and try to register it, nothing should happen.

Now, let's try it for real:

1. `yarn build`/`npm run build`
2. `cd public`
3. Bring up an http server there.
4. Open the networks tab.
5. Write an invalid email address and try to register it, you should see the GA request.

> If you are on mac or have Python installed, you can run this:
> 
> ```sh
> python -m SimpleHTTPServer
> ```

### CODE REVIEW

File: src/common/ga.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { waitForGlobal } from './utils';
+
+let client;
+const getClient = () => {
```
Comment (joelalejandro) [inline src/common/ga.js]: async/await would make this less... nesty

File: src/common/ga.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  }
+
+  return waitForGlobal('ga').then(gaClient => {
+    client = gaClient;
+    return client;
+  });
+};
+
+export const trackEvent = (category, action, label, value) => {
+  if (process.env.NODE_ENV === 'production') {
```
Comment (joelalejandro) [inline src/common/ga.js]: Let's avoid if-blocks wrapping entire functions. It's preferable to early exit.

File: src/components/AddEmailForm/AddEmailForm.jsx [hunk 3]
```diff
@@ -30,41 +31,53 @@ const _AddEmailForm = ({ emailError, isAdding, wasSaved, add }) => {
   );
 
   const formErrors = emailError ? [emailError] : errors;
+  if (formErrors && formErrors.length) {
```
Comment (joelalejandro) [inline src/components/AddEmailForm/AddEmailForm.jsx]: To avoid the &&, trinaries and stuff, I'd do:

```const formErrors = [...(emailError || errors || [])];```

File: src/components/AddEmailForm/AddEmailForm.jsx [hunk 4]
```diff
@@ -30,41 +31,53 @@ const _AddEmailForm = ({ emailError, isAdding, wasSaved, add }) => {
...
+        <form className={styles.form} onSubmit={onSubmit}>
+          <Input
+            placeholder="Tu e-mail"
+            value={email}
+            onChange={e => setEmail(e.target.value)}
+            hasError={!!formErrors}
+            disabled={isAdding}
+            autoFocus
+          />
+          {!!formErrors
```
Comment (joelalejandro) [inline src/components/AddEmailForm/AddEmailForm.jsx]: The above suggestion would remove the need for this check.

File: src/common/utils.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+};
+
+export const waitForGlobal = (name, intervalTime = 600, attempts = 5) => {
+  const defer = deferred();
+  let currentAttempt = 0;
+  let interval = setInterval(() => {
+    currentAttempt++;
+    if (global[name]) {
+      defer.resolve(global[name]);
+      clearInterval(interval);
```
Comment (joelalejandro) [inline src/common/utils.js]: clearInterval() could be avoided by spawning a setTimeout() while the condition isn't met.

File: src/common/ga.js (hunk 1 above)
Comment (homer0) [reply src/common/ga.js]: Agree

File: src/common/ga.js (hunk 2 above)
Comment (homer0) [reply src/common/ga.js]: Agree

File: src/components/AddEmailForm/AddEmailForm.jsx (hunk 3 above)
Comment (homer0) [reply src/components/AddEmailForm/AddEmailForm.jsx]: To be honest, I would always prefer a ternary over a triple short circuit inside a spread; but I can't argue with the fact that normalizing it wouldn't be better for the multiple places where it's used.

File: src/common/utils.js (hunk 5 above)
Comment (homer0) [reply src/common/utils.js]: Agree

File: src/components/AddEmailForm/AddEmailForm.jsx (hunk 3 above)
Comment (joelalejandro) [reply src/components/AddEmailForm/AddEmailForm.jsx]: At the very least, we could initialize it as an array for default value.
